Automattic/wp-calypso
# FSE: Handle Gutenboarding launch button

#### Changes proposed in this Pull Request

Part of https://github.com/Automattic/wp-calypso/pull/42847.
Companion to https://github.com/Automattic/wp-calypso/pull/42849.

Details in https://github.com/Automattic/wp-calypso/pull/42847 and D44235-code.

#### Testing instructions

Sandbox `widgets.wp.com` and `example.wordpress.com` (Simple site).

* Test D44235-code in isolation. It should result in no changes in the block editor.
* Test in combination with D44258-code. It should result in no changes in the block editor.

1. https://wordpress.com/block-editor should work as before.
1. Starting from [`/new`](https://wordpress.com/new), follow the flow through site creation to the editor.
1. In the editor, the publish button should be replaced with "Launch."
1. The launch button should link to a URL like https://wordpress.com/start/new-launch?siteSlug=example.wordpress.com&source=editor

All of the testing is to ensure there are no regressions.
